CMakeLists.txt

```diff
@@ -2773,26 +2773,26 @@ if(CLANG_COLORDIAG)
 endif()
 
 # Clang Sanitizers
-set(CLANG_SANITIZERS "")
-option(CLANG_ASAN "Clang Address Sanitizer" OFF)
-if(CLANG_ASAN)
-  list(APPEND CLANG_SANITIZERS "address")
-endif()
-option(CLANG_UBSAN "Clang Undefined Behaviour Sanitizer" OFF)
-if(CLANG_UBSAN)
-  list(APPEND CLANG_SANITIZERS "undefined")
-endif()
-option(CLANG_TSAN "Clang Thread Sanitizer" OFF)
-if(CLANG_TSAN)
-  list(APPEND CLANG_SANITIZERS "thread")
-endif()
-if(NOT CLANG_SANITIZERS STREQUAL "")
-  if(NOT LLVM_CLANG)
-    message(FATAL_ERROR "Clang Sanitizers are only available when using Clang.")
-  endif()
-  list(JOIN CLANG_SANITIZERS "," CLANG_SANITZERS_JOINED)
-  target_compile_options(mixxx-lib PUBLIC -fsanitize=${CLANG_SANITZERS_JOINED})
-  target_link_options(mixxx-lib PUBLIC -fsanitize=${CLANG_SANITZERS_JOINED})
+set(SANITIZERS "")
+option(SANITIZE_ADDRESS "Address Sanitizer" OFF)
+if(SANITIZE_ADDRESS)
+  list(APPEND SANITIZERS "address")
+endif()
+option(SANITIZE_UNDEFINED "Clang Undefined Behaviour Sanitizer" OFF)
+if(SANITIZE_UNDEFINED)
+  list(APPEND SANITIZERS "undefined")
+endif()
+option(SANITIZE_THREAD "Clang Thread Sanitizer" OFF)
+if(SANITIZE_THREAD)
+  list(APPEND SANITIZERS "thread")
+endif()
+if(NOT SANITIZERS STREQUAL "")
+  if(NOT (LLVM_CLANG OR GNU_GCC))
+    message(FATAL_ERROR "Sanitizers are only available on Clang or GCC")
```

Comment:
ASAN is also available in MSVC: https://learn.microsoft.com/en-us/cpp/sanitizers/asan

Reply:
yes, but the commandline syntax is a little different and I didn't want to go KISS (also I would have had no way to test the MSVC code path). Feel free to implement yourself in a follow-up PR though.